src/foam/dao/pg/PostgresDAO.java

```diff
@@ -0,0 +1,368 @@
+        .collect(Collectors.toList());
+  }
+
+  public Sink select_(X x, Sink sink, long skip, long limit, Comparator order, Predicate predicate) {
+    if ( sink == null ) {
+      sink = new ListSink();
+    }
+
+    Connection c = null;
+    PreparedStatement stmt = null;
+    ResultSet resultSet = null;
+
+    try {
+      c = connectionPool.getConnection();
+
+      // select all if predicate is null, else use predicate
+      if ( predicate == null ) {
+        StringBuilder builder = sb.get()
+            .append("select * from ")
+            .append(table);
+        stmt = c.prepareStatement(builder.toString());
+      } else {
+        stmt = c.prepareStatement(predicate.createStatement(table));
+      }
+
+      resultSet = stmt.executeQuery();
+      while ( resultSet.next() ) {
+        sink.put(createFObject(resultSet), null);
+      }
+
+      return sink;
+    } catch (Exception e) {
+      e.printStackTrace();
+      return null;
+    } finally {
+      if ( resultSet != null )
+        try { resultSet.close(); } catch (SQLException ignored) {}
+      if ( stmt != null )
+        try { stmt.close(); } catch (SQLException ignored) {}
+      if ( c != null )
+        try { c.close(); } catch (SQLException ignored) {}
+    }
+  }
+
+  @Override
+  public FObject remove_(X x, FObject o) {
+    Connection c = null;
+    PreparedStatement stmt = null;
+
+    try {
+      c = connectionPool.getConnection();
+      StringBuilder builder = sb.get()
+          .append("delete from ")
+          .append(table)
+          .append(" where id = ?");
+
+      stmt = c.prepareStatement(builder.toString());
+      // TODO: add support for non-numbers
+      stmt.setLong(1, ((Number) o.getProperty("id")).longValue());
+
+      int removed = stmt.executeUpdate();
+      if ( removed == 0 ) {
+        throw new SQLException("Error while removing.");
+      }
+
+      return o;
+    } catch (Exception e) {
+      e.printStackTrace();
+      return null;
+    } finally {
+      if ( stmt != null )
+        try { stmt.close(); } catch (SQLException ignored) {}
+      if ( c != null )
+        try { c.close(); } catch (SQLException ignored) {}
+    }
+  }
+
+  @Override
+  public FObject find_(X x, Object o) {
+    Connection c = null;
+    PreparedStatement stmt = null;
+    ResultSet resultSet = null;
+
+    try {
+      c = connectionPool.getConnection();
+      StringBuilder builder = sb.get()
+          .append("select * from ")
+          .append(table)
+          .append(" where id = ?");
+
+      stmt = c.prepareStatement(builder.toString());
+      // TODO: add support for non-numbers
+      stmt.setLong(1, ((Number) o).longValue());
+      resultSet = stmt.executeQuery();
+      if ( ! resultSet.next() ) {
+        // no rows
+        return null;
+      }
+
+      return createFObject(resultSet);
+    } catch (Exception e) {
+      e.printStackTrace();
+      return null;
+    } finally {
+      if ( resultSet != null )
```

Comment:
Since this pattern repeats, it could be a standard method called closeAll(resultSet, stmt, c).

Reply:
Added a helper function